### DISCUSSION TITLE
genes in the corresponding k means cluster in P2G Heatmaps

### DISCUSSION THREAD

**Discussion by sylestiel** [GreenleafLab/ArchR#632] [Questions / Documentation] [answered]
In order to show a few representative genes corresponding to each k mean cluster in the heatmap of peak-to-gene links how can it be achieved? So far I have been able to obtain the following values.

seqnames | start | end | width | strand | value | FDR
-- | -- | -- | -- | -- | -- | --

Although I can annotate the regions to get the corresponding gene name I do not know which ones correspond to the appropriate k mean generated cluster.

Any thoughts?

**Comment by rcorces** [comment, marked as answer]
Does this help? https://github.com/GreenleafLab/ArchR/issues/132

The `returnMatrices` parameter should give you what you want:
https://www.archrproject.com/reference/plotPeak2GeneHeatmap.html

**Comment by sylestiel** [comment]
Thank you @ rcorces. It did.

**Comment by sylestiel** [comment]
@rohitrrj 

FYI check out the comment from rcorces about finding the kmeans Id corresponding to the P2G links.

Hope it helps you too!
